### PR TITLE
py-halo: new port

### PR DESCRIPTION
**Pull request by bashu**
#### Description

Beautiful spinners for terminal, IPython and Jupyter 

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 12.6.2 21G320 x86_64
Xcode 14.2 14C18

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: python/py-halo/Portfile [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+platforms           {darwin any}
+
+maintainers         {@bashu gmail.com:bashu.was.here} openmaintainer
+
+description         Beautiful terminal spinners in Python
+long_description    {*}${description}
+
+homepage            https://github.com/manrajgrover/halo
+
+distname            halo-${version}
```
Comment (reneeotten) [inline python/py-halo/Portfile:21]: is this required - I *think* it's the default already?

File: python/py-halo/Portfile [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+checksums           rmd160  2a67ba23ee8883c2ba3c4870a64ce2201d2d4819 \
+                    sha256  7b67a3521ee91d53b7152d4ee3452811e1d2a6321975137762eb3d70063cc9d6 \
+                    size    11666
+
+python.versions     37 38 39 310
```
Comment (reneeotten) [inline python/py-halo/Portfile:27]: perhaps only add recent version at this time unless there are other packages that depend on older subports. I'd recommend "310 311" only.

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: python/py-halo/Portfile (hunk 2 above)
Comment (bashu) [reply python/py-halo/Portfile]: fixed

File: python/py-halo/Portfile (hunk 1 above)
Comment (bashu) [reply python/py-halo/Portfile]: fixed

**Comment by reneeotten** [comment]
it appears that there are missing dependencies for the py311 subport....

**Comment by bashu** [comment]
> it appears that there are missing dependencies for the py311 subport....

Here they are:

https://github.com/macports/macports-ports/pull/17140
https://github.com/macports/macports-ports/pull/17141
https://github.com/macports/macports-ports/pull/17142

**Pull request merged**